lib/rig_inbound_gateway/api_proxy/validations.ex

```diff
@@ -88,6 +88,16 @@ defmodule RigInboundGateway.ApiProxy.Validations do
 
   # ---
 
+  @spec validate_endpoint_path(Api.endpoint()) :: error_list_t()
+  def validate_endpoint_path(endpoint) do
+    errors = validate_string(endpoint, "path") ++ validate_string(endpoint, "path_regex")
```

Comment:
what if both are valid? Indeed, I'm not sure what should happen in case _both_ properties are set. If you are, please add a test that documents/explains that behavior.

Reply:
No idea, what was your intention when adding `path_regex`?

Reply:
The idea was to [keep `path` simple](https://github.com/Accenture/reactive-interaction-gateway/blob/master/lib/rig_inbound_gateway/api_proxy/api.ex#L15) and not introduce a breaking change there, but [add `path_regex` as a more powerful alternative](https://github.com/Accenture/reactive-interaction-gateway/blob/master/lib/rig_inbound_gateway/api_proxy/api.ex#L19).

Turns out, the way I've implemented it [allows to use both at the same time, and it'll just pick whatever matches first](https://github.com/Accenture/reactive-interaction-gateway/blob/master/lib/rig_inbound_gateway/api_proxy/api.ex#L100). Not sure that was intended, I don't quite recall.

To sum up, perhaps it's best to simply allow both be present and mention that if that's the case, only one of them needs to match. Additionally, we should consider deprecating `path` to remove it with 3.0.

Reply:
Hmm, that sounds a bit unpredictable and dangerous to me. Would rather prefer either one or another as it anyway probably doesn't make sense to use both at the same time.

[Here](https://github.com/Accenture/reactive-interaction-gateway/issues/278) we can get rid of everything deprecated and path as well if we want.

Reply:
> Hmm, that sounds a bit unpredictable and dangerous to me. Would rather prefer either one or another as it anyway probably doesn't make sense to use both at the same time.

Unfortunately, changing this behavior now would be breaking. Then again: I don't expect anyone using both directives, since it doesn't make sense, like you said.